src/pyhf/readxml.py

```diff
@@ -5,7 +5,6 @@
     IO,
     Callable,
     Iterable,
-    List,
     MutableMapping,
```

Comment:
We can do this because in Python 3.9+ (where we're going) typing has improved/been migrated in from `typing`?

Reply:
no, because python 3.8 has `from __future__ import annotations` which lets us do that, but ruff wasn't warning about this yet because it didn't look at the combination of a missing future annotations and the fact that the target was 3.8 - so it assumed it was 3.8, rather than targeting `< 3.9`.